pkg/kubelet/kubelet.go

```diff
@@ -544,6 +544,12 @@ func NewMainKubelet(kubeCfg *kubeletconfiginternal.KubeletConfiguration,
 		klet.cloudResourceSyncManager = cloudresource.NewSyncManager(klet.cloud, nodeName, klet.nodeStatusUpdateFrequency)
 	}
 
+	// Set the noop metrics provider for all reflectors that will be created
+	// after this point, so that watch-based managers (if used) will not
+	// blow the Kubelet memory usage if there are huge churn of secrets and/or
+	// configmaps attached to pods running on this node.
+	cache.SetReflectorMetricsProvider(manager.NoopMetricsProvider{})
```

Comment:
Actually it won't work, because SetReflectorMetricsProvider is using sync.Once underneath (so once initialized, we can't change it).

That said, I realized that I don't understand why this is happening, because the:
- metrics factory is defaulted to noop
- the prometheus metrics need to be explicitly linked and initialized (as they are in apiserver e.g.:
https://github.com/kubernetes/kubernetes/blob/d1031b18fc83146dd8418c17df289d09d10d29e5/cmd/kube-apiserver/app/server.go#L81 )

So I'm not sure what the exact problem actually is...